src/main/java/com/o19s/es/ltr/logging/LoggingFetchSubPhase.java

```diff
@@ -201,7 +203,7 @@ void doLog(Query query, List<HitLogConsumer> loggers, IndexSearcher searcher, Se
         }
 
         private void rebuild() {
-            List<Map<String, Object>> ini = new ArrayList<>(set.size());
+            List<Map<String, Object>> ini = new ArrayList<>(set.size() + 1);
```

Comment:
nit: add a small comment to explain why you over allocate here, might help the reader to understand that extraLogging will be added when needed